### ISSUE TITLE
Selecting null in a single select should deselect the current item

### ISSUE THREAD

**Issue by Artur-** [vaadin/framework8-issues#294]
In Vaadin 7 you can do

```
ComboBox contract = ...;
contract.select(bean.getContract());
```

which will synchronize the selection in `contract` to the `contract` in the bean either to a contract or an empty selection (if null).

Doing this right now requires something like

```
if (bean.getContract() == null) {
    contract.getSelectedItem().ifPresent(contract::deselect);
} else {
    contract.select(bean.getContract());
}
```

**Comment by jdahlstrom** [comment]
```
contract.setSelectedItem(bean.getContract());
```

**Comment by Artur-** [comment]
Why should the contract of select change? It certainly makes migration much harder if you need to check every invocation of select to see if it can select null

**Comment by jdahlstrom** [comment]
What should `select(null)` do in multiselect? or should it just behave differently in singleselect? :( Would be nice to be able to update bad old APIs :( :( :(

**Comment by Legioth** [comment]
1) ComboBox is a single select. Why should its API be sub-par just because other components might be multi selects?
2) ComboBox is top typically used in the same way as a TextField, not as a Grid. Therefore, it would make sense if its primary API is centered around setting a value instead of a selection.
3) We don't break old APIs just for the sake of breaking. But when there is a clear benefit of changing an API, then it must be done in a way that causes compilation errors instead of runtime exceptions for legacy code.

**Comment by pleku** [comment]
SHOULD HAVE, but not necessary for FW8.

**Comment by pleku** [comment]
This has been implemented for 8.0.0.beta1
